lib/testScenario/swaggerAnalyzer.ts

```diff
@@ -0,0 +1,238 @@
+import { JSONPath } from "jsonpath-plus";
+import { inject, injectable } from "inversify";
+import _ from "lodash";
+import { inversifyGetInstance, TYPES } from "../inversifyUtils";
+import { FileLoaderOption } from "../swagger/fileLoader";
+import { JsonLoader, JsonLoaderOption } from "../swagger/jsonLoader";
+import { SwaggerLoader, SwaggerLoaderOption } from "../swagger/swaggerLoader";
+import { Path, SwaggerExample, SwaggerSpec } from "../swagger/swaggerTypes";
+import { AjvSchemaValidator } from "../swaggerValidator/ajvSchemaValidator";
+import { SchemaValidator } from "../swaggerValidator/schemaValidator";
+import { allOfTransformer } from "../transform/allOfTransformer";
+import { getTransformContext, TransformContext } from "../transform/context";
+import { discriminatorTransformer } from "../transform/discriminatorTransformer";
+import { noAdditionalPropertiesTransformer } from "../transform/noAdditionalPropertiesTransformer";
+import { nullableTransformer } from "../transform/nullableTransformer";
+import { pureObjectTransformer } from "../transform/pureObjectTransformer";
+import { referenceFieldsTransformer } from "../transform/referenceFieldsTransformer";
+import { resolveNestedDefinitionTransformer } from "../transform/resolveNestedDefinitionTransformer";
+import { xmsPathsTransformer } from "../transform/xmsPathsTransformer";
+import { applyGlobalTransformers, applySpecTransformers } from "../transform/transformer";
+import { traverseSwaggerAsync } from "../transform/traverseSwagger";
+import { getProvider } from "../util/utils";
+import { setDefaultOpts } from "../swagger/loader";
+
+export interface SwaggerAnalyzerOption
+  extends FileLoaderOption,
+    JsonLoaderOption,
+    SwaggerLoaderOption {
+  swaggerFilePaths?: string[];
+  noExternalDependencyResourceType?: boolean;
+  filerTopLevelResourceType?: boolean;
+}
+
+export interface ExampleDependency {
+  exampleFilePath: string;
+  exampleName: string;
+  apiVersion: string;
+  resourceProvider: string;
+  resourceType: string;
+  internalDependency: Dependency;
```

Comment:
Should this be array too?

Reply:
internalDependency is analyzed by path. So it should be only one dependencyChain.